src/Microsoft.AspNetCore.Cors/Infrastructure/CorsService.cs

```diff
@@ -151,8 +151,7 @@ public virtual void EvaluatePreflightRequest(HttpContext context, CorsPolicy pol
             {
                 foreach (var requestHeader in requestHeaders)
                 {
-                    if (!CorsConstants.SimpleRequestHeaders.Contains(requestHeader, StringComparer.OrdinalIgnoreCase) &&
```

Comment:
I hadn't noticed `SimpleRequestHeaders` being used as an additional allow list. @Eilon \ @javiercn should we leave the existing behavior since it's breaking? I'm all for removing magical behavior (and this is entirely magical since it's second guessing the user), but seems like it's less cost for us to keep it for now.

The specific scenario that's affected by this is when `AllowAnyHeader = false`, but the configured allow headers does not include one of the following - `Origin`, `Accept`, `Accept-Language`, or `Content-Language`. CORS will start rejecting these requests where it previously did.

Reply:
@pranavkm  How's that if I may ask? line 154 checks the configured headers.
Also you might be talking about the filterin from (removed) lines 220 to 231 which made that rejecting occur or am I wrong?

Reply:
@pranavkm I agree, then this line should stay in here as this is kinda spec behavior isn't it?

Reply:
I agree with removing magic. But we should post an announcement for this.

Reply:
So what is the play @pranavkm? Reverting the change or keeping it like this?

Reply:
Keeping it as is seems is fine. I'll get this merged in as soon as I get a change to draft an announcement.

Reply:
Does CORS reject requests if it doesn't contain any of the 'Simple headers' or if it doesn't contain all the 'Simple headers'?